src/listpack.h

```diff
@@ -31,6 +33,13 @@ typedef struct {
     uint32_t slen;
     /* When integer is used, 'sval' is NULL, and lval holds the value. */
     long long lval;
+
+    /* These variables are used internally by lpInsert() */
+    int enctype;
+    uint64_t enclen;
+    unsigned char intenc[LP_MAX_INT_ENCODING_LEN];
+    unsigned long backlen_size;
+    unsigned char backlen[LP_MAX_BACKLEN_SIZE];
```

Comment:
i'm sure if we should put these in here instead of adding a new structure like listpackInsertEntry.

Reply:
sure, if confusing, we can introduce a new struct

Reply:
It looks like `listpackInsertEntry` is wrong, it's used to store the encoded data in the middle of insertion.
perhaps we want another struct, and it's visible for inside.

Reply:
I think it is not possible hide it from outside but maybe we can put all these fields into another struct?:

```
typedef struct {
    /* When string is used, it is provided with the length (slen). */
    unsigned char *sval;
    uint32_t slen;
    /* When integer is used, 'sval' is NULL, and lval holds the value. */
    long long lval;

    struct listpackInsertEntry internal; /* Internally used */
} listpackEntry;

Reply:
but i didn't see `listpackInsertEntry` was used out listpack.c.
am i missing somethimg?

Reply:
`listpackInsertEntry` is not used elsewhere. I meant we can introduce it and add it into `listpackEntry`.  

Maybe I misunderstood you. Do you want to make it totally internal to listpack.c? I think it is not possible without some allocation (either on stack or heap). 

e.g. lpBatchInsert() is called 10 items. How do you allocate 10 listpackInsertEntry items?

Reply:
yes, i meant to allocate a dynamic listpackInsertEntry array inside `lpinsertBatch`.

Reply:
This is what I was trying to avoid :) I'm not sure if performance cost worth it. Also, it will be regression for the existing code. Maybe we can selectively do it. 

e.g. have stack allocated listpackInsertEntry arr[5] in lpInsertBatch().
If more than 5 entries are given, allocate on heap. 

Still not sure if this is better.

Reply:
a static array also works.
i just don't think we should let these encoding data be perceived outside.